### PR TITLE
Add "set up universal links" cookbook recipe

### PR DESCRIPTION
**Pull request by chunhtai**
fixes https://github.com/flutter/website/issues/7913

requires https://github.com/flutter/codelabs/pull/1443/files

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/cookbook/navigation/set-up-universal-links.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+path: /cookbook/navigation/passing-data
+next:
+title: Delete data on the internet
+path: /cookbook/networking/delete-data
+js:
+- defer: true
+  url: https://dartpad.dev/inject_embed.dart.js
+---
+
+<?code-excerpt path-base="cookbook/navigation/deeplink"?>
```
Comment (chunhtai) [inline src/cookbook/navigation/set-up-universal-links.md:15]: Not sure what this line actually does, just copy over from other markdown

File: src/cookbook/navigation/set-up-universal-links.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+Next, navigate to <strong>Runner-> Sign & Signature -> + Capability -> Associated Domains</strong>
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-associated-domains.png" alt="Xcode associated domains screenshot"/>
+</noscript>
+
+In the <strong>Associated Domains</strong> section click +
+and enter <strong>applinks:\<web domain\></strong>.
+This recipe uses the domain name: chunhtai.github.io.
```
Comment (chunhtai) [inline src/cookbook/navigation/set-up-universal-links.md:115]: A note to myself, replace `chunhtai.github.io` with more appropriate one

File: src/cookbook/navigation/set-up-universal-links.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+You should replace it with your own domain name.
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-add-associated-domains.png" alt="Xcode add associated domains screenshot"/>
```
Comment (chunhtai) [inline src/cookbook/navigation/set-up-universal-links.md:119]: This img also need to be updated

File: src/cookbook/navigation/set-up-universal-links.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+If everything is set up correctly, the Flutter application
+launches and displays the details screen:
+
+<noscript>
+  <img src="/assets/images/docs/cookbook/set-up-universal-links-simulator-deeplinked.png" alt="Simulator screenshot"/>
+</noscript>
+
+[alternate mode section]: https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_associated-domains?language=objc
+[sample repo]: https://github.com/chunhtai/chunhtai.github.io
```
Comment (chunhtai) [inline src/cookbook/navigation/set-up-universal-links.md:198]: A note to myself, need to replace this with more appropriate github page account

File: src/cookbook/navigation/set-up-universal-links.md (hunk 1 above)
Comment (domesticmouse) [reply src/cookbook/navigation/set-up-universal-links.md:15]: That expects a matching Dart codebase at the link. The cookbook collection of samples is copied in from https://github.com/flutter/codelabs/tree/main/cookbook

File: src/cookbook/navigation/set-up-universal-links.md (hunk 2 above)
Comment (domesticmouse) [reply src/cookbook/navigation/set-up-universal-links.md:115]: @johnpryan I think this needs to be under https://flutter.github.io/samples/# somehow. Preferences?

**Comment by domesticmouse** [comment]
I'm curious how the CI passed with new unbacked `code-excepts`. @khanhnwin?

File: src/cookbook/navigation/set-up-universal-links.md (hunk 2 above)
Comment (chunhtai) [reply src/cookbook/navigation/set-up-universal-links.md:115]: This is for github page. and the url to the AASA file has to be under
https://flutter.github.io/.well-known/apple-app-site-association.

I think we need a new account for this.

**Comment by chunhtai** [comment]
> I'm curious how the CI passed with new unbacked `code-excepts`. @khanhnwin?

I added the code in.

File: src/cookbook/navigation/set-up-universal-links.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
+---
+title: Set up universal links for iOS
+description: How to animate a widget from one screen to another
```
Comment (sfshaza2) [inline src/cookbook/navigation/set-up-universal-links.md:3]: Update this. :)

File: src/cookbook/navigation/set-up-universal-links.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+contains scheme, host, and path and leads users to a specific
+screen of an app.
+
+A universal link is a deeplink with http or https scheme that
+is exclusive to Apple devices.
+
+
+## Prerequisite
+1. A mac with Flutter SDK and Xcode installed
+2. A web domain
```
Comment (sfshaza2) [inline src/cookbook/navigation/set-up-universal-links.md:27]: We don't typically include this for a recipe.

```suggestion
```

We do typically write a short intro and list the broad steps. For example:
```
The “swipe to dismiss” pattern is common in many mobile apps. For example, when writing an email app, you might want to allow a user to swipe away email messages to delete them from a list.

Flutter makes this task easy by providing the [Dismissible](https://api.flutter.dev/flutter/widgets/Dismissible-class.html) widget. Learn how to implement swipe to dismiss with the following steps:

1. Create a list of items.
2. Wrap each item in a Dismissible widget.
3. Provide “leave behind” indicators.
```

File: src/cookbook/navigation/set-up-universal-links.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+If you don’t own a web domain and still want to test the
+universal link, consider using [Firebase Hosting][] or
+[GitHub Pages][] as a temporary solution. These instructions
+use GitHub Pages.
+
+
+## 1. Write a Flutter application
```
Comment (sfshaza2) [inline src/cookbook/navigation/set-up-universal-links.md:35]:
```suggestion
## 1. Customize a Flutter application
```

File: src/cookbook/navigation/set-up-universal-links.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+First, write a flutter application that can handle an incoming
+URL. Start by creating a new application with flutter create.
+
+```shell
+$ flutter create universallinks
+```
+
+This example uses [go_router][] to handle the routing. The
+go_router is a first party package maintained by Flutter
```
Comment (sfshaza2) [inline src/cookbook/navigation/set-up-universal-links.md:45]:
```suggestion
go_router is a first party package maintained by the Flutter team
```

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
@chunhtai, this is my first pass. It needs more work to meet the cookbook recipe format.

**Review by atsansone — CHANGES_REQUESTED** [review]
@chunhtai : I provided some feedback for consistency with Google Style and a few questions.

File: src/cookbook/navigation/set-up-universal-links.md [hunk 9]
```diff
@@ -82,52 +82,56 @@ final router = GoRouter(
 
 ## 2. Adjust iOS build settings
 
-Launch Xcode. Open the `ios/Runner.xcworkspace`
-file inside the project’s Flutter folder. Navigate to the
-`info.Plist` file:
+1. Launch Xcode.
+2. Open the `ios/Runner.xcworkspace` file inside the project’s `Flutter` folder.
```
Comment (johnpryan) [inline src/cookbook/navigation/set-up-universal-links.md:86]: I think it's in the `ios` folder:

```suggestion
2. Open the `ios/Runner.xcworkspace` file inside the project’s `ios` folder.
```

File: src/cookbook/navigation/set-up-universal-links.md [hunk 10]
```diff
@@ -82,52 +82,56 @@ final router = GoRouter(
...
+3. Navigate to the `Info` Plist file.
```
Comment (johnpryan) [inline src/cookbook/navigation/set-up-universal-links.md:87]:
```suggestion
3. Navigate to the `Info` Plist file. in the `ios/Runner` folder.
```

File: src/cookbook/navigation/set-up-universal-links.md [hunk 11]
```diff
@@ -82,52 +82,56 @@ final router = GoRouter(
...
+4. In the `Info` property list, add a row.
+5. Set the key to `FlutterDeepLinkingEnabled` with a
 `Boolean` value set to `YES`.
 
-<noscript>
-  <img src="/assets/images/docs/cookbook/set-up-universal-links-flutterdeeplinkingenabled.png" alt="flutter deeplinking enabled screenshot"/>
-</noscript>
+    <noscript>
+      <img src="/assets/images/docs/cookbook/set-up-universal-links-flutterdeeplinkingenabled.png" alt="flutter deeplinking enabled screenshot"/>
+    </noscript>
```
Comment (johnpryan) [inline src/cookbook/navigation/set-up-universal-links.md:99]: I think it's better to use markdown for these:

```suggestion
![flutter deeplinking enabled screenshot](/assets/images/docs/cookbook/set-up-universal-links-flutterdeeplinkingenabled.png)
```

File: src/cookbook/navigation/set-up-universal-links.md [hunk 12]
```diff
@@ -82,52 +82,56 @@ final router = GoRouter(
...
+7. Click **Sign & Signature**.
+8. Click **+ Capability** to add a new domain.
+9. Click **Associated Domains**.
 
-<noscript>
-  <img src="/assets/images/docs/cookbook/set-up-universal-links-associated-domains.png" alt="Xcode associated domains screenshot"/>
-</noscript>
+    <noscript>
+      <img src="/assets/images/docs/cookbook/set-up-universal-links-associated-domains.png" alt="Xcode associated domains screenshot"/>
+    </noscript>
```
Comment (johnpryan) [inline src/cookbook/navigation/set-up-universal-links.md:108]:
```suggestion
![Xcode associated domains screenshot](/assets/images/docs/cookbook/set-up-universal-links-associated-domains.png)
```

File: src/cookbook/navigation/passing-data.md [hunk 13]
```diff
@@ -6,7 +6,7 @@ prev:
   path: /cookbook/navigation/returning-data
 next:
   title: Delete data on the internet
-  path: /cookbook/networking/delete-data
+  path: /cookbook/navigation/set-up-universal-links
```
Comment (domesticmouse) [inline src/cookbook/navigation/passing-data.md:9]: This path doesn't seem to match the title

Comment (chunhtai) [reply src/cookbook/navigation/passing-data.md:9]: nice catch!

**Comment by domesticmouse** [comment]
> @domesticmouse Do you know how can i get the ci check for stable to pass? It looks like the beta works.

Looking at the build log it's failing on the snippets check, which only runs on stable. Please see the README for instructions on how to update the snippets

**Comment by chunhtai** [comment]
@johnpryan I end up have to go back to img tag since that gives me control over css style size

**Comment by chunhtai** [comment]
A gentle ping, this is ready for another look

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Comment by dntandan** [comment]
@chunhtai thank you for your PR. Your PR documentation helped a lot today for solving the Universal Links issue on flutter.

Thank you!

File: src/cookbook/navigation/set-up-universal-links.md [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+
+## 2. Adjust iOS build settings
+
+1. Launch Xcode.
+2. Open the `ios/Runner.xcworkspace` file inside the project’s `ios` folder.
+3. Navigate to the `Info` Plist file in the `ios/Runner` folder.
+
+    <img src="/assets/images/docs/cookbook/set-up-universal-links-info-plist.png" alt="Xcode info.Plist screenshot" width="100%" />
+
+4. In the `Info` property list, add a row.
```
Comment (atsansone) [inline src/cookbook/navigation/set-up-universal-links.md:92]: **question:** What do you do to add a row? What do you press?

File: src/cookbook/navigation/set-up-universal-links.md [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+5. Set the key to `FlutterDeepLinkingEnabled` with a
```
Comment (atsansone) [inline src/cookbook/navigation/set-up-universal-links.md:93]: **question:** How do you set the key? Do you fill in a field? Which field?

Don't rely on the screenshot to communicate your ask. There could be folks that aren't using a visual browser or other issues, like language barriers. (Images can't be automatically translated.)

**Review by atsansone — APPROVED** [review]
LGTM % two questions. Answer the questions and you can merge without further review from me.

File: src/cookbook/navigation/set-up-universal-links.md (hunk 14 above)
Comment (chunhtai) [reply src/cookbook/navigation/set-up-universal-links.md]: added more detailed instructions

File: src/cookbook/navigation/set-up-universal-links.md (hunk 15 above)
Comment (chunhtai) [reply src/cookbook/navigation/set-up-universal-links.md]: added more details instructions

**Pull request merged**